lib/httplib/httplib.go

```diff
@@ -371,3 +372,21 @@ func (r *ResponseStatusRecorder) Hijack() (net.Conn, *bufio.ReadWriter, error) {
 	}
 	return h.Hijack()
 }
+
+// TODO(kimlisa): DELETE IN v19.0 (csrf)
+// Deprecated: do not use (only here to support backwards compat for old plugin endpoints)
+// WithCSRFProtection ensures that request to unauthenticated API is checked against CSRF attacks
+func WithCSRFProtection(fn HandlerFunc) httprouter.Handle {
```

Comment:
Similarly, maybe we don't need to export this at all and we can move the form field check directly into `WithAuthCookieAndCSRF` since that seems to be the only place this is used.